### PR TITLE
0003 automatic mode enhancement

### PR DESCRIPTION
**Pull request by ycheng**
push automatic mode further as in development phase, fast turn around will be useful!

Certain modification might not be the best way to make it work. Please kindly comment if you have better suggestions.

For file "/cdrom/dvd-automatic", I prefer name usb-automatic, since we use usb stick much more than dvd.
If you also think that work, I'd like to modify all dvd to usb in the source code.

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -7,6 +7,7 @@ dell-recovery (1.66) UNRELEASED; urgency=medium
   [ Yuan-Chen Cheng ]
   * sort disk name by type and size.
   * properly show disk with rst attribute.
+  * skip this step if we run in automatic-oem-config mode.
```
Comment (superm1) [inline debian/changelog:10]: This commit message should be clearer what step is being skipped

File: ubiquity/dell-recovery.py [hunk 2]
```diff
@@ -125,6 +125,8 @@ def prepare(self, unfiltered=False):
         """Prepares the debconf plugin"""
         destination = self.db.get('dell-recovery/destination')
         self.ui.set_type(destination)
+        if 'UBIQUITY_AUTOMATIC' in os.environ:
```
Comment (superm1) [inline ubiquity/dell-recovery.py:128]: I think you should leave a comment above this explaining this is meant for development purposes

**Comment by superm1** [comment]
Rather than more loose files on the recovery partition, what do you think about adding a new valid type to `dell-recovery/recovery_type`?  Perhaps `dev`?  And when this is preseeded then do all the same things.

File: ubiquity/dell-recovery.py (hunk 2 above)
Comment (ycheng) [reply ubiquity/dell-recovery.py:128]: Re-check the code, and found another place to modify to make it more nature.

File: debian/changelog (hunk 1 above)
Comment (ycheng) [reply debian/changelog]: please check the new pushed branch, thanks.

**Comment by ycheng** [comment]
Per test today, found the patch in cf32d56 does not have stable behaviour. Sometimes it failed to move foward.
I think it's because at the time go_forward() is call, the gui loop might not run yet. Need to check a better way to implement it.

**Comment by ycheng** [comment]
No problem, I'll use one preseed value to do that. Due to my work load, it could take some more time.

**Comment by ycheng** [comment]
this PR has been replaced by https://github.com/dell/dell-recovery/pull/118, which just merged.
Given so, close this one.

**Pull request closed without merging**